spec/unit/views/components/scopes_spec.rb

```diff
@@ -0,0 +1,143 @@
+# frozen_string_literal: true
+require "rails_helper"
+
+RSpec.describe ActiveAdmin::Views::Scopes do
+  describe "the scopes list" do
+    let(:collection) { Post.all }
+    let(:active_admin_config) { ActiveAdmin.register(Post) }
+
+    let(:assigns) do
+      {
+        active_admin_config: active_admin_config,
+        collection_before_scope: collection
+      }
+    end
+
+    let(:helpers) do
+      helpers = mock_action_view
+
+      allow(helpers.request)
+        .to receive(:path_parameters)
+        .and_return(controller: "admin/posts", action: "index")
+
+      helpers
+    end
+
+    let(:configured_scopes) do
+      [
+        ActiveAdmin::Scope.new(:all),
+        ActiveAdmin::Scope.new(:published) { |posts| posts.where.not(published_date: nil) }
+      ]
+    end
+
+    let(:scope_options) do
+      { scope_count: true }
+    end
+
+    let(:scopes) do
+      scopes_to_render = configured_scopes
+      options = scope_options
+
+      render_arbre_component assigns, helpers do
+        insert_tag(ActiveAdmin::Views::Scopes, scopes_to_render, options)
+      end
+    end
+
+    before do
+      allow(ActiveAdmin::AsyncCount).to receive(:new).and_call_original
+    end
+
+    around do |example|
+      with_resources_during(example) { active_admin_config }
+    end
+
+    it "renders the scopes component" do
+      html = Capybara.string(scopes.to_s)
+      expect(html).to have_css("div.scopes")
+
+      configured_scopes.each do |scope|
+        expect(html).to have_css("a[href='/admin/posts?scope=#{scope.id}']")
+      end
+    end
+
+    context "when scopes are configured to query their counts asynchronously" do
+      let(:configured_scopes) do
+        [
+          ActiveAdmin::Scope.new(:all, nil, show_count: :async),
+          ActiveAdmin::Scope.new(:published, nil, show_count: :async) { |posts| posts.where.not(published_date: nil) }
+        ]
+      end
+
+      it "raises an error when ActiveRecord async_count is unavailable", unless: Post.respond_to?(:async_count) do
+        expect { scopes }.to raise_error(NoMethodError, %r{async_count})
```

Comment:
Would be easy to display a nicer error message in this scenario? 
Something like "scope `show_count: :async` is not available in Rails X.Y.Z. This Rails version doesn't support async_count". 
I know this feature is `opt-in` but I'm wondering if we can explain in a better way the cause of the problem. I want to avoid future issues asking why they get NoMethodError.
Of course, if it's easy.

Reply:
Done in [f4205c3](https://github.com/activeadmin/activeadmin/pull/8394/commits/f4205c30a48a531a627b639bb34d12f8c4870fb1)